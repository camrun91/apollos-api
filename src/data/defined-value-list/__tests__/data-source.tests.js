import ApollosConfig from '@apollosproject/config'
import DefinedValueList from '../data-source'

ApollosConfig.loadJs({
    ROCK: {
        API_URL: 'https://apollosrock.newspring.cc/api',
        API_TOKEN: 'some-rock-token',
        IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
    },
})

describe('Defined Value List', () => {
    it('gets a defined value from a valid guid or integer id', async () => {
        const dataSource = new DefinedValueList()
        const id = '999'

        dataSource.get = jest.fn(() => Promise.resolve([{ Id: 999, Value: 'Foo' }]))

        const result = dataSource.getByIdentifier(id)

        expect(result).toMatchSnapshot()
    })
})