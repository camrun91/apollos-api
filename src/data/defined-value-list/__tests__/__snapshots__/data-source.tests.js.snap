// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Defined Value List gets a defined value from a valid guid or integer id 1`] = `Promise {}`;
